server/services/DeckService.js

```diff
@@ -310,10 +310,21 @@ class DeckService {
 
         try {
             decks = await db.query(
-                'SELECT *, CASE WHEN "WinCount" + "LoseCount" = 0 THEN 0 ELSE (CAST("WinCount" AS FLOAT) / ("WinCount" + "LoseCount")) * 100 END AS "WinRate" FROM ( ' +
+                'SELECT *, ' +
+                    'CASE WHEN "WinCount" + "LoseCount" = 0 THEN 0 ELSE (CAST("WinCount" AS FLOAT) / ("WinCount" + "LoseCount")) * 100 END AS "WinRate", ' +
+                    'CASE WHEN "BeginnerWinCount" + "BeginnerLoseCount" = 0 THEN 0 ELSE (CAST("BeginnerWinCount" AS FLOAT) / ("BeginnerWinCount" + "BeginnerLoseCount")) * 100 END AS "BeginnerWinRate", ' +
+                    'CASE WHEN "CasualWinCount" + "CasualLoseCount" = 0 THEN 0 ELSE (CAST("CasualWinCount" AS FLOAT) / ("CasualWinCount" + "CasualLoseCount")) * 100 END AS "CasualWinRate", ' +
+                    'CASE WHEN "CompetitiveWinCount" + "CompetitiveLoseCount" = 0 THEN 0 ELSE (CAST("CompetitiveWinCount" AS FLOAT) / ("CompetitiveWinCount" + "CompetitiveLoseCount")) * 100 END AS "CompetitiveWinRate" ' +
+                    'FROM ( ' +
                     'SELECT d.*, u."Username", e."ExpansionId" as "Expansion", (SELECT COUNT(*) FROM "Decks" WHERE "Name" = d."Name") AS DeckCount, ' +
                     '(SELECT COUNT(*) FROM "Games" g JOIN "GamePlayers" gp ON gp."GameId" = g."Id" WHERE g."WinnerId" = $1 AND gp."DeckId" = d."Id") AS "WinCount", ' +
-                    '(SELECT COUNT(*) FROM "Games" g JOIN "GamePlayers" gp ON gp."GameId" = g."Id" WHERE g."WinnerId" != $1 AND g."WinnerId" IS NOT NULL AND gp."PlayerId" = $1 AND gp."DeckId" = d."Id") AS "LoseCount" ' +
+                    '(SELECT COUNT(*) FROM "Games" g JOIN "GamePlayers" gp ON gp."GameId" = g."Id" WHERE g."WinnerId" != $1 AND g."WinnerId" IS NOT NULL AND gp."PlayerId" = $1 AND gp."DeckId" = d."Id") AS "LoseCount", ' +
```

Comment:
This should all be fetched in one join and then grouped. If you can't figure out the SQL for grouping it, then grouping it in code is acceptable